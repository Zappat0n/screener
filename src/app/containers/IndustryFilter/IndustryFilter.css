.filter .content{
  margin: 1rem;
  font: normal normal 700 0.8rem/0.8rem 'Open Sans', sans-serif;
}

.filter .content select {
  font: normal normal 700 0.8rem/0.8rem 'Open Sans', sans-serif;
}
