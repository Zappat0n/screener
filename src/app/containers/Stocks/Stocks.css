.stocks table {
  border-collapse: collapse;
}

th,
td {
  border: 1px solid lightgray;
  padding: 0.3rem 0.5rem;
}
