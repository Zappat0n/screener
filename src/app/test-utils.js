import React from 'react';
import { createStore, combineReducers } from 'redux';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import indexReducer from './reducers/indexSlice';
import stockReducer from './reducers/stockSlice';
import profileReducer from './reducers/profileSlice';
import filterReducer from './reducers/filterSlice';

const rootReducer = combineReducers({
  indexes: indexReducer, stocks: stockReducer, profile: profileReducer, filter: filterReducer,
});

const initialState = {
  stocks: {
    AAPL: {
      price: 100,
      companyName: 'Apple Inc',
      changes: 1,
      sector: 'Technology',
    },
  },
  profile: {
    data: {
      image: '',
      companyName: 'Microsoft',
      price: 100,
      changes: 1,
      mktCap: 1000000,
      range: '0-100',
      country: 'US',
    },
  },
};

const renderComponent = (component) => render(
  <Provider store={createStore(rootReducer, initialState)}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>,
);

export * from '@testing-library/react';
export { renderComponent };
