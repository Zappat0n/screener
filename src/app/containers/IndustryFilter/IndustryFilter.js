import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import './IndustryFilter.css';

const IndustryFilter = (props) => {
  const { handleChange } = props;
  const stocks = useSelector((state) => state.stocks);
  const filters = ['All'];

  const createFilters = () => {
    Object.values(stocks).forEach((data) => {
      if (data.sector && !filters.includes(data.sector)) {
        filters.push(data.sector);
      }
    });
  };
  createFilters();

  return (
    <div className="filter">
      <div className="content">
        Sector:
        {'  '}
        <select
          className="industry-filter"
          onChange={(event) => handleChange(event.target.value)}
        >
          {filters.map((sector) => (
            <option key={sector} value={sector}>
              {sector}
            </option>
          ))}
        </select>
      </div>
    </div>
  );
};

IndustryFilter.propTypes = {
  handleChange: PropTypes.func.isRequired,
};

export default IndustryFilter;
