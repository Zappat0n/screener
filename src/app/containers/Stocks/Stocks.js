import { useSelector, useDispatch } from 'react-redux';
import { unwrapResult } from '@reduxjs/toolkit';
import { useParams } from 'react-router-dom';
import Stock from '../Stock/Stock';
import IndustryFilter from '../IndustryFilter/IndustryFilter';
import { updateFilter } from '../../reducers/filterSlice';
import {
  clearStocks, refreshStocks, setIndex, updateStocks,
} from '../../reducers/stockSlice';
import indexComponents from '../../data/components';

import './Stocks.css';

const Stocks = () => {
  const { id } = useParams();
  const components = useSelector((state) => state.stocks);
  const filter = useSelector((state) => state.filter);
  const dispatch = useDispatch();

  const initialize = () => {
    dispatch(updateFilter('All'));
    dispatch(clearStocks());
    dispatch(setIndex(id));
    dispatch(refreshStocks(indexComponents[id].join(',')))
      .then(unwrapResult)
      .then((originalPromiseResult) => {
        dispatch(updateStocks(originalPromiseResult));
      });
  };

  if (components.index !== id) {
    initialize();
  }

  const createStocks = () => {
    if (components) {
      return Object.keys(components).filter((quote) => (
        (components[quote].sector === filter.value || filter.value === 'All')) && quote !== 'index').map(
        (quote) => <Stock key={quote} ticker={quote} />,
      );
    }
    return [];
  };

  const changeFilter = (sector) => {
    dispatch(updateFilter(sector));
  };

  return (
    <>
      <div className="stocks">
        <IndustryFilter handleChange={changeFilter} />
        <table>
          <thead>
            <tr>
              <th>Name</th>
              <th>Price</th>
              <th>Change</th>
              <th>Sector</th>
            </tr>
          </thead>
          <tbody>
            {createStocks()}
          </tbody>
        </table>
      </div>
    </>
  );
};

export default Stocks;
