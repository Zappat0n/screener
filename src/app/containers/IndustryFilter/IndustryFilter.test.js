import React from 'react';
import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import IndustryFilter from './IndustryFilter';
import Stocks from '../Stocks/Stocks';
import { renderComponent, rendererForComponent } from '../../test-utils';

test('Renders Filter correctly', () => {
  renderComponent(
    <>
      <Stocks />
      <IndustryFilter handleChange={() => {}} />
    </>,
  );
  expect(screen.getAllByText('Technology')[0]).toBeInTheDocument();
  expect(screen.getAllByText('All')[0]).toBeInTheDocument();
});

test('Index Match SnapShot', () => {
  const tree = rendererForComponent(
    <>
      <Stocks />
      <IndustryFilter handleChange={() => {}} />
    </>,
  );
  expect(tree).toMatchSnapshot();
});
