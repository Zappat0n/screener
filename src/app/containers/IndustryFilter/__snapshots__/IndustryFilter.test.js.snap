// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Match SnapShot 1`] = `
Array [
  <div
    className="stocks"
  >
    <div
      className="filter"
    >
      <div
        className="content"
      >
        Sector:
          
        <select
          className="industry-filter"
          onChange={[Function]}
        >
          <option
            value="All"
          >
            All
          </option>
          <option
            value="Technology"
          >
            Technology
          </option>
        </select>
      </div>
    </div>
    <table>
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Price
          </th>
          <th>
            Change
          </th>
          <th>
            Sector
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="stock"
        >
          <td>
            <a
              className="link "
              href="/stock/AAPL"
              onClick={[Function]}
            >
              Apple Inc
            </a>
          </td>
          <td>
            <a
              className="link "
              href="/stock/AAPL"
              onClick={[Function]}
            >
              100
            </a>
          </td>
          <td>
            <a
              className="link colorBlue"
              href="/stock/AAPL"
              onClick={[Function]}
            >
              1%
            </a>
          </td>
          <td>
            <a
              className="link "
              href="/stock/AAPL"
              onClick={[Function]}
            >
              Technology
            </a>
          </td>
        </tr>
        <tr
          className="stock"
        >
          <td>
            <a
              className="link "
              href="/stock/MSFT"
              onClick={[Function]}
            >
              Microsoft
            </a>
          </td>
          <td>
            <a
              className="link "
              href="/stock/MSFT"
              onClick={[Function]}
            >
              110
            </a>
          </td>
          <td>
            <a
              className="link colorBlue"
              href="/stock/MSFT"
              onClick={[Function]}
            >
              NaN%
            </a>
          </td>
          <td>
            <a
              className="link "
              href="/stock/MSFT"
              onClick={[Function]}
            />
          </td>
        </tr>
        <tr
          className="stock"
        >
          <td>
            <a
              className="link "
              href="/stock/GOOG"
              onClick={[Function]}
            >
              Google
            </a>
          </td>
          <td>
            <a
              className="link "
              href="/stock/GOOG"
              onClick={[Function]}
            >
              120
            </a>
          </td>
          <td>
            <a
              className="link colorBlue"
              href="/stock/GOOG"
              onClick={[Function]}
            >
              NaN%
            </a>
          </td>
          <td>
            <a
              className="link "
              href="/stock/GOOG"
              onClick={[Function]}
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <div
    className="filter"
  >
    <div
      className="content"
    >
      Sector:
        
      <select
        className="industry-filter"
        onChange={[Function]}
      >
        <option
          value="All"
        >
          All
        </option>
        <option
          value="Technology"
        >
          Technology
        </option>
      </select>
    </div>
  </div>,
]
`;
